lib/jack_tokenizer.rb

```diff
@@ -11,12 +11,21 @@ def initialize(io)
   LETTERS = ("a".."z").to_a + ("A".."Z").to_a
 
   def has_more_tokens?
-    while WHITESPACES.include?(@input[@index])
-      @index += 1
-    end
+    loop do
+      old_index = @index
+
+      while WHITESPACES.include?(@input[@index])
+        @index += 1
+      end
+
+      if @input[@index] == "/" && @input[@index+1] == "/"
+        @index = @input.index("\n", @index) || @input.length
+
+      elsif @input[@index] == "/" && @input[@index+1] == "*"
```

Comment:
This code grew organically and it works, but now we’re starting to copy it around, maybe we could clean it up a little. Instead of comparing two adjacent characters to `"/"` and `"*"` individually, could we just pull out the two-character substring starting at `@index` and compare it to `"/*"` directly? I think that’d be both more concise and easier to understand. (Ditto for checking for `"//"` earlier.)